src/components/FilterableProductTable.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+import SearchBar from './SearchBar';
+import ProductTable from './ProductTable';
+import 'bulma/css/bulma.css';
+
+export default class FilterableProductTable extends Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            products: this.props.products,
+            inStock: []
+        }
+    }
+
+    filterProducts = searchQuery => {
+        const filteredProducts = searchQuery ? this.state.products.filter(product => product.name.toUpperCase().includes(searchQuery.toUpperCase())) : this.props.products;
+    
+        this.setState({
+            products: filteredProducts
+        })
+    }
+
+    showInStock = checkboxValue => {
+        const inStockProducts = checkboxValue ? this.props.products : this.state.products.filter(product => product.stocked == true);
+
+        this.setState({
+            products: inStockProducts
+        })
+    }
+
+    render() {
+        let shownProducts = this.state.products;
```

Comment:
Everything looks great, althouth this line of code doesn't save you much time, you could just passed the "`this.state.products`" directly as props :P